File: NotificationAdmin/src/app/services/notification.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, catchError, throwError } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class NotificationService {
  private apiUrl = 'http://localhost:5125/api/Notification';

  constructor(private http: HttpClient) {}

  sendNotification(title: string, body: string, sentBy: string): Observable<Notification> {
    const payload = { title, body, sentBy };
    return this.http.post<Notification>(`${this.apiUrl}/send`, payload).pipe(
      catchError(err => {
        console.error('Error sending notification:', err);
        return throwError(() => err);
      })
    );
  }

}

File: JupiterTask/JupiterTask/Core/Entities/Notification.cs
namespace JupiterTask.Core.Entities
{
    public class Notification
    {
        public int Id { get; set; }

        public required string Title { get; set; }

        public required string Body { get; set; }

        public DateTime CreatedAt { get; set; }

        public required string SentBy { get; set; }

        public bool IsSent { get; set; }
    }
}
